upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -156,8 +165,8 @@ func createMetaData(dir string, vmName string) error {
 	return nil
 }
 
-func createISO(changes *AttachISO, startupStr string, dir string, dnsServer string) (string, error) {
-	err := createUserData(startupStr, dir, dnsServer)
+func createISO(changes *AttachISO, startupStr string, dir string, dnsServer string, vmUUID string) (string, error) {
```

Comment:
How about changing signature to- func (change *AttachISO) createISO(....), and same for createUserData. 

I have done this for my changes so that the number of arguments passed to createISO and createUserData in check. Otherwise with your and my changes we will sendup passing 6-7 arguments to these methods.